src/context.cpp

```diff
@@ -943,6 +952,54 @@ namespace librealsense
             throw std::runtime_error( "Could not find a stream that supports option " + name );
         }
 
+        void add_processing_block( std::string filter_name )
+        {
+            auto & current_filters = get_software_recommended_proccesing_blocks();
+            
+            if( processing_block_exists( current_filters.get_recommended_processing_blocks(), filter_name ) )
+                return; // Already created by another stream of this sensor
+
+            create_processing_block( filter_name );
+        }
+
+        bool processing_block_exists( processing_blocks const & blocks, std::string const & block_name )
+        {
+            for( auto & block : blocks )
+                if( block_name.compare( block->get_info( RS2_CAMERA_INFO_NAME ) ) == 0 )
+                    return true;
+
+            return false;
+        }
+
+        void create_processing_block( std::string & filter_name )
+        {
+            auto & current_filters = get_software_recommended_proccesing_blocks();
+
+            if( filter_name.compare( "Depth Huffman Decoder" ) == 0 )
```

Comment:
Consider removing the "==0" and adding"!" instead. e.g:
if( !filter_name.compare( "Depth Huffman Decoder" ) )

Reply:
I think that using `!` is less readable, because compare returns 0 when it is true. !compare reads like it does not compare.

Reply:
your style - ok